File: OsmAnd-java/src/main/java/net/osmand/binary/RouteDataBundle.java
package net.osmand.binary;

import net.osmand.router.RouteDataResources;

public class RouteDataBundle extends StringBundle {

	private RouteDataResources resources;

	public RouteDataBundle(RouteDataResources resources) {
		this.resources = resources;
	}

	public RouteDataBundle(RouteDataResources resources, StringBundle bundle) {
		super(bundle.getMap());
		this.resources = resources;
	}

	@Override
	public StringBundle newInstance() {
		return new RouteDataBundle(resources);
	}

	public RouteDataResources getResources() {
		return resources;
	}
}

File: OsmAnd-shared/src/commonMain/kotlin/net/osmand/shared/routing/RouteDataBundle.kt
package net.osmand.shared.routing

import net.osmand.shared.util.StringBundle

/** A [StringBundle] that carries the route wide state a segment needs while it is read or written. */
class RouteDataBundle : StringBundle {

	private val resources: RouteDataResources

	constructor(resources: RouteDataResources) : super() {
		this.resources = resources
	}

	constructor(resources: RouteDataResources, bundle: StringBundle) : super(bundle) {
		this.resources = resources
	}

	override fun newInstance(): StringBundle = RouteDataBundle(resources)

	fun getResources(): RouteDataResources = resources
}
